### PR TITLE
Revised chain_sim() documentation

### LINKED ISSUE #4: clarify documentation for @return in chain_sim / change behavior for `tree` + `serial` combinations
The documentation for return value in `chain_sim` suggests that all combinations of `serial` and `tree` are allowed, but function complains if `tree=FALSE` with a `serial` function defined.

This suggests either a documentation or behavior bug.

Relative to behavior option, I have a case where I don't need `id`, `generation`, `ancestor` information, but do need `time` and am memory constrained, so disregarding those would be useful.

### PR DESCRIPTION
**Pull request by jamesmbaazam**
This pull request will update the current chain_sim() documentation to resolve #4 and address #18.

### CODE REVIEW

File: DESCRIPTION [hunk 1]
```diff
@@ -1,20 +1,27 @@
...
+Version: 0.1.0
 Authors@R: c(
-    person("Sebastian", "Funk", email = "sebastian.funk@lshtm.ac.uk", role = c("aut", "cre")),
-    person("Zhian N.", "Kamvar", email = "zkamvar@gmail.com", role = c("ctb")),
-    person("Flavio", "Finger", email = "flavio.finger@epicentre.msf.org", role = c("aut"))
-    )
-Description: Provides methods to analyse and simulate the size and length of branching processes with an arbitrary offspring distribution. These can be used, for example, to analyse the distribution of chain sizes or length of infectious disease outbreaks, as discussed in Farrington et al. (2003) <doi:10.1093/biostatistics/4.2.279>.
+    person("Sebastian", "Funk", , "sebastian.funk@lshtm.ac.uk", role = c("aut", "cre")),
+    person("Zhian N.", "Kamvar", , "zkamvar@gmail.com", role = "ctb"),
+    person("Flavio", "Finger", , "flavio.finger@epicentre.msf.org", role = "aut")
```
Comment (sbfnk) [inline DESCRIPTION:7]: Please add yourself

File: R/simulate.r [hunk 2]
```diff
@@ -1,39 +1,87 @@
...
-##'   chain), `ancestor` (the ID of the ancestor of each element) and
-##'   `generation`. 
-##' @author Sebastian Funk
-##' @export
-##' @examples
-##' chain_sim(n=5, "pois", "size", lambda=0.5)
+#' Simulate transmission chains using a branching process
+#' @description \code{chain_sim()} is a stochastic simulator for generating 
+#' transmission chain data given information on the offspring distribution, 
+#' serial interval, time since the first case, e.t.c.
```
Comment (sbfnk) [inline R/simulate.r:4]:
```suggestion
#' serial interval, time since the first case, etc. 
```

File: R/simulate.r [hunk 3]
```diff
@@ -1,39 +1,87 @@
...
-##' chain_sim(n=5, "pois", "size", lambda=0.5)
+#' Simulate transmission chains using a branching process
+#' @description \code{chain_sim()} is a stochastic simulator for generating 
+#' transmission chain data given information on the offspring distribution, 
+#' serial interval, time since the first case, e.t.c. 
+#' @param n Number of simulations to run.
+#' @param offspring Offspring distribution: a character string corresponding to
+#'   the R distribution function (e.g., "pois" for Poisson, where
+#'   \code{\link{rpois}} is the R function to generate Poisson random numbers) 
+#' @param stat String; Statistic to calculate.
```
Comment (sbfnk) [inline R/simulate.r:9]:
```suggestion
#' @param stat String; Statistic to calculate. Can be one of
```

File: R/simulate.r [hunk 4]
```diff
@@ -1,39 +1,87 @@
...
+#' \itemize{
+#'   \item "size": the total number of offspring.
+#'   \item "length": the total number of ancestors. 
+#' }
+#' @param infinite A size or length above which the simulation results should be 
+#' set to `Inf`. Defaults to `Inf`.
```
Comment (sbfnk) [inline R/simulate.r:15]:
```suggestion
#' set to `Inf`. Defaults to `Inf`, resulting in no results ever set to `Inf`
```

File: R/simulate.r [hunk 5]
```diff
@@ -1,39 +1,87 @@
...
+#' \itemize{
+#'  \item{A vector of sizes/lengths (if \code{tree == FALSE} OR serial
+#'   interval function not specified, since that implies \code{tree == FALSE})}, or 
+#'   \item {a data frame with 
+#'   columns `n` (simulation ID), `time` (if the serial interval is given) and 
+#'   (if \code{tree == TRUE}), `id` (a unique ID within each simulation for each 
+#'   individual element of the chain), `ancestor` (the ID of the ancestor of each 
+#'   element), and `generation`.}
+#' }
+#' @author Sebastian Funk
```
Comment (sbfnk) [inline R/simulate.r:34]: add yourself

File: R/simulate.r [hunk 6]
```diff
@@ -1,39 +1,87 @@
...
+#' @export
+#' @details 
+#' `chain_sim()` either returns a vector or a data.frame. The output is either a 
+#' vector if serial is not provided, which automatically sets \code{tree = FALSE},
```
Comment (sbfnk) [inline R/simulate.r:38]:
```suggestion
#' vector if `serial` is not provided, which automatically sets \code{tree = FALSE},
```

File: R/simulate.r [hunk 7]
```diff
@@ -1,39 +1,87 @@
...
+#' or a data.frame, which means that `serial` was provided as a function. When `serial`
```
Comment (sbfnk) [inline R/simulate.r:39]:
```suggestion
#' or a `data.frame`, which means that `serial` was provided as a function. When `serial`
```

File: R/simulate.r [hunk 8]
```diff
@@ -1,39 +1,87 @@
...
+#' is provided, it means \code{tree = TREE} automatically. However, setting
```
Comment (sbfnk) [inline R/simulate.r:40]:
```suggestion
#' is provided, it means \code{tree = TRUE} automatically. However, setting 
```

File: vignettes/introduction.Rmd [hunk 9]
```diff
@@ -1,14 +1,23 @@
 ---
 title: "Analysing chain statistics using branching process models"
 author: "Sebastian Funk"
```
Comment (sbfnk) [inline vignettes/introduction.Rmd:3]: add yourself

**Review by sbfnk — CHANGES_REQUESTED** [review]
Really nice! Just left a few minor comments. Can the file `references.bib.sav` be removed?

File: README.Rmd [hunk 10]
```diff
@@ -0,0 +1,31 @@
...
+)
+```
+[![Travis-CI Build Status](https://travis-ci.org/sbfnk/bpmodels.svg?branch=master)](https://travis-ci.org/sbfnk/bpmodels)
+[![Appveyor Build Status](https://ci.appveyor.com/api/projects/status/y37i8x0wo9o8s2wf?svg=true)](https://ci.appveyor.com/project/sbfnk/bpmodels)
+[![codecov](https://codecov.io/github/sbfnk/bpmodels/branch/master/graphs/badge.svg)](https://codecov.io/github/sbfnk/bpmodels) 
+```{r setup, include=FALSE}
+knitr::opts_chunk$set(echo = TRUE)
+```
+
+`bpmodels` is an R package for simulating epidemics using branching processes. The package provides functions for fitting the following models:
```
Comment (sbfnk) [inline README.Rmd:22]: Is there something missing here (which models)? Also at the moment we don't have any functions for fitting, right? Just simulating and calculating likelihoods.

File: .gitignore [hunk 11]
```diff
@@ -26,3 +26,9 @@ vignettes/*.pdf
 *.knit.md
 # Shiny token, see https://shiny.rstudio.com/articles/shinyapps.html
 rsconnect/
+/doc/
+/Meta/
+
+.Rbuildignore
+
+*.Rproj
```
Comment (sbfnk) [inline .gitignore:34]: Maybe add `*.bib.sav` and remove `references.bib.sav` as I don't think these are needed in the repo.

**Review by sbfnk — CHANGES_REQUESTED** [review]
Looks great! Just a couple of more commetns.

File: README.Rmd (hunk 10 above)
Comment (jamesmbaazam) [reply README.Rmd]: That has been resolved.

File: .gitignore (hunk 11 above)
Comment (jamesmbaazam) [reply .gitignore:34]: Resolved.

**Comment by pratikunterwegs** [comment]
Thanks for tagging @jamesmbaazam. I see the R CMD check workflows are failing but that seems to be a (possibly minor) issue with vignette building. Would be good to get all checks passing and then I can take a look.

After a quick glance though, does this commit https://github.com/epiverse-trace/bpmodels/pull/14/commits/1cfd4303557befc0f0f121c540eb380d22eeaab0 represent a merge of `main` into `james`? I would suggest restting this branch to before this commit, and rebasing on `main` as we discussed on Slack, and fixing merge conflicts during that rebase.

The sequence of commands would be:

```sh
git checkout james
git reset --hard HEAD~1

# make sure local main is updated with the remote
git rebase main
# fix conflicts

# force push the 'james' branch
git push -f
```

**Comment by jamesmbaazam** [comment]
> does this commit [1cfd430](https://github.com/epiverse-trace/bpmodels/commit/1cfd4303557befc0f0f121c540eb380d22eeaab0) represent a merge of `main` into `james`

Yes, it does. Currently resetted and rebased. Thanks for the help.

**Pull request merged**